src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -104,25 +104,33 @@ public async Task SpecializeAsync()
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
 
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
-            if (_workerRuntime != null && languageWorkerChannel != null)
+            if (UsePlaceholderChannel(_workerRuntime, languageWorkerChannel != null))
             {
-                // TODO: https://github.com/Azure/azure-functions-host/issues/4534 Don't kill non read-only processes once linked issue is resolved
-                if (_environment.FileSystemIsReadOnly())
-                {
-                    _logger.LogDebug("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                    await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
-                }
-                else
-                {
-                    _logger.LogDebug("Shutting down placeholder worker. Worker is not compatible for runtime: {runtime}", _workerRuntime);
-                    // If we need to allow file edits, we should shutdown the webhost channel on specialization.
-                    await ShutdownChannelIfExistsAsync(_workerRuntime, languageWorkerChannel.Id);
-                }
+                _logger.LogDebug("Loading environment variables for runtime: {runtime}", _workerRuntime);
+                await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+            }
+            else
+            {
+                _logger.LogDebug("Shutting down placeholder worker. Worker is not compatible for runtime: {runtime}", _workerRuntime);
+                // If we need to allow file edits, we should shutdown the webhost channel on specialization.
+                await ShutdownChannelIfExistsAsync(_workerRuntime, languageWorkerChannel.Id);
             }
+
             _shutdownStandbyWorkerChannels();
             _logger.LogDebug("Completed language worker channel specialization");
         }
 
+        internal bool UsePlaceholderChannel(string workerRuntime, bool channelExists)
```

Comment:
add unit tests just for UsePlaceholderChannel. Simpler to verify conditions

Reply:
I think I did in WebHostLanguageWorkerManagerTests, is that what you're thinking of too?

Reply:
just tests that verify true or false when calling UsePlaceholderChannel